azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -723,16 +723,19 @@ void closeMBeans() {
    */
   @VisibleForTesting
   void shutdown() {
-    logger.info("Shutting down the pod");
-    final int execId = this.flowRunner.getExecutionId();
-    while (!this.flowFuture.isDone()) {
-      // This should not happen immediately as submitFlowRunner is a blocking call.
-      try {
-        Thread.sleep(100);
-      } catch (final InterruptedException e) {
-        logger.error(String.format("The sleep while waiting for execution : %d to finish was interrupted",
-            execId));
+    logger.info("Shutting down the container");
+    if (this.flowRunner != null) {
+      final int execId = this.flowRunner.getExecutionId();
+      while (!this.flowFuture.isDone()) {
+        // This should not happen immediately as submitFlowRunner is a blocking call.
+        try {
+          Thread.sleep(100);
```

Comment:
Perhaps in the future we should try a wait() implementation that is more precise.